spark-indexer/src/main/java/org/uniprot/store/spark/indexer/chebi/mapper/ChebiRelatedIdsMapper.java

```diff
@@ -24,15 +25,35 @@
     @Override
     public Iterator<Tuple2<Long, Long>> call(ChebiEntry entry) throws Exception {
         final Long entryId = Long.parseLong(entry.getId());
+        List<Tuple2<Long, Long>> result = new ArrayList<>();
         if (Utils.notNullNotEmpty(entry.getRelatedIds())) {
-            return entry.getRelatedIds().stream()
+            entry.getRelatedIds().stream()
                     .map(ChebiEntry::getId)
                     .map(Long::parseLong)
                     .map(relatedId -> new Tuple2<>(relatedId, entryId))
-                    .collect(Collectors.toList())
-                    .iterator();
-        } else {
-            return Collections.emptyIterator();
+                    .forEach(result::add);
         }
+
+        if (Utils.notNullNotEmpty(entry.getMajorMicrospecies())) {
+            entry.getMajorMicrospecies().stream()
+                    .map(ChebiEntry::getId)
+                    .map(Long::parseLong)
+                    .flatMap(majorMicrospecieId -> createMajorMicroespeciesRelation(entryId, majorMicrospecieId))
+                    .forEach(result::add);
+        }
+        return result.iterator();
+    }
```

Comment:
we can move the codes inside two if blocks to two different methods with a self explanatory name.